### PR TITLE
Core/Spells: Gathering no longer fails after patch 3.1.0

### LINKED ISSUE #21038: Core/Spells: Players are able to fail while gathering
**Description:** Players are able to fail while gathering

**Current behaviour:**

Players can fail when attempting to skin a mob, mine a node or pick a flower

**Expected behaviour:**

Per the 3.1.0 patch notes (http://www.wowhead.com/patchnotes=3.1.0):

You can no longer fail when Mining, Herbing, and Skinning.

**Steps to reproduce the problem:**

Attempt to gather enough skins, ores or herbs and you will see failed attempts.

Due to the random nature of failing when gathering it will only happen once in a while. You will have a greater chance at reproducing the issue on 'orange' level resources.

**Branch(es):** 3.3.5 (not tested on master)

**TC rev. hash/commit:**

TrinityCore rev. 97f33dc 2017-11-11 20:27:30 +0100 (3.3.5 branch) (Unix, RelWithDebInfo, Static) (authserver)

**Operating system:** Ubuntu 16.04LTS

### PR DESCRIPTION
**Pull request by Ovahlord**
**Changes proposed:**

-  removed the chance of failing at gathering materials such as herbalism since it has been removed with patch 3.1.0:

Gathering Skills

>     Toughness, Master of Anatomy, and Lifeblood now have level requirements equal to the level required for the corresponding skill rank: Apprentice (1), Journeyman (1), Expert (10), Artisan (25), Master (40), Grand Master (55).
>     You can no longer fail when Mining, Herbing, and Skinning.

**Target branch(es):** both

- [x] 3.3.5
- [x] master

**Issues addressed:** (really) Closes #21038

### CODE REVIEW

**Comment by tkrokli** [comment]
BTW, if you would like to see issue #21038 automatically closed when this PR gets merged, you would need to add a commit message text (2 lines below the commit title) _in your first commit_ saying `Closes #21038` (could also be done in a squash or merge of the commits) - but I think it can be done on in the merge process as well. Anyway, I'm just mentioning this in case this github feature is something you would like to try out for yourself.

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -5400,13 +5394,13 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
                 if (res != SPELL_CAST_OK)
                     return res;
 
-                // chance for fail at orange mining/herb/LockPicking gathering attempt
+                // chance for fail at lockpicking attempt
                 // second check prevent fail at rechecks
                 if (skillId != SKILL_NONE && (!m_selfContainer || ((*m_selfContainer) != this)))
                 {
-                    bool canFailAtMax = skillId != SKILL_HERBALISM && skillId != SKILL_MINING;
+                    bool canFailAtMax = skillId == SKILL_LOCKPICKING;
```
Comment (Treeston) [inline src/server/game/Spells/Spell.cpp]: What's the purpose of this change? What `skillId`s other than `_HERBALISM`, `_MINING` and `_LOCKPICKING` does this affect?

Comment (Shauren) [reply src/server/game/Spells/Spell.cpp]: Easier to understand for readers.

Comment (Shauren) [reply src/server/game/Spells/Spell.cpp]: Easier to understand for future readers

**Review by Shauren — APPROVED** [review]

File: src/server/game/Spells/Spell.cpp (hunk 1 above)
Comment (Treeston) [reply src/server/game/Spells/Spell.cpp]: Ah, so there's no other skill IDs? Alright.

**Pull request merged**